Rego/DefenderConfig.rego

```diff
@@ -421,6 +421,62 @@ tests[{
 # MS.DEFENDER.4.1v1
 #--
 
+# Return set of content info types in basic rules
+InfoTypeMatches(Rule) := ContentTypes if {
+    Rule.IsAdvancedRule == false
+    ContentTypes := {Content.name | Content = Rule.ContentContainsSensitiveInformation[_]}
+}
+
+# Return set of content info types in advanced rules
+InfoTypeMatches(Rule) := ContentTypes if {
+    Rule.IsAdvancedRule == true
+    RuleText := replace(replace(Rule.AdvancedRule, "rn", ""), "'", "\"")
+
+    # Split string to keep line length intact
+    TypesRegex := concat("", [`(U.S. Social Security Number \(SSN\))|`,
+                                `(U.S. Individual Taxpayer Identification `,
+                                `Number \(ITIN\))|`,
+                                `(Credit Card Number)`
+                            ]
+                        )
+    ContentNames := regex.find_n(TypesRegex, RuleText, -1)
+
+    ContentTypes := { name | name := ContentNames[_] }
+}
+
+SensitiveInfoTypes(PolicyName) := MatchingInfoTypes if {
+    InfoTypes := {  "U.S. Social Security Number (SSN)",
+                    "U.S. Individual Taxpayer Identification Number (ITIN)",
+                    "Credit Card Number"
+                }
+
+    ContentTypeSets := { Types | Rule := input.dlp_compliance_rules[_]
+                                    Rule.Disabled == false
+                                    Rule.ParentPolicyName == PolicyName
+                                    Types := InfoTypeMatches(Rule)
+                       }
+    # Flatten set of sets of content types
+    ContentTypes := union(ContentTypeSets)
+
+    MatchingInfoTypes := InfoTypes & ContentTypes
+}
+
+# Return set of policy names that contain all sensitive info types in rules
+SensitiveInfoPolicies[Policy.Name] {
+    Policy := input.dlp_compliance_policies[_]
+
+    # Inspect rules for sensitive info types
+    InfoTypes := SensitiveInfoTypes(Policy.Name)
+
+    Conditions := [ "U.S. Social Security Number (SSN)" in InfoTypes,
+                    "U.S. Individual Taxpayer Identification Number (ITIN)" in InfoTypes,
+                    "Credit Card Number" in InfoTypes]
+
+    count([Condition | Condition = Conditions[_]; Condition == true]) == 3
+    Policy.Enabled = true
+    Policy.Mode == "Enable"
```

Comment:
SensitiveInfoPolicies is used in all Policies part of the 4 group. But I did not see unit test for line 476 in 4.1 & 4.3. I did not see unit test for failing line 477 in 4.1 & 4.2.